### PR TITLE
Actix web telemetry

### PR DESCRIPTION
**Pull request by IvanPleshkov**
### All Submissions:

* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you lint your code locally using ``cargo fmt`` command prior to submission?
3. [ ] Have you checked your code using ```cargo clippy``` command?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/actix/actix_telemetry.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    actix_web::dev::forward_ready!(service);
+
+    fn call(&self, request: ServiceRequest) -> Self::Future {
+        let future = self.service.call(request);
+        let telemetry_data = self.telemetry_data.clone();
+        Box::pin(async move {
+            let response = future.await?;
+            let status = response.response().status().as_u16();
+            telemetry_data.lock().unwrap().add_response(status as usize);
```
Comment (agourlay) [inline src/actix/actix_telemetry.rs:37]: Is it possible to use an async. `Mutex` instead?
I am a bit concerned about blocking and the funky stuffs possibly happening with guards regarding async. suspension.

Comment (generall) [reply src/actix/actix_telemetry.rs:37]: +1, let's use tokio mutex

Comment (IvanPleshkov) [reply src/actix/actix_telemetry.rs:37]: But I'll change to tokio mutex if it using tokio doesn't require a ton of additional code

Comment (IvanPleshkov) [reply src/actix/actix_telemetry.rs]: Fixed. Using async tokio mutex for telemetry collector and parking_lot mutex for builder

File: src/common/user_telemetry.rs [hunk 2]
```diff
@@ -73,13 +76,19 @@ pub struct UserTelemetryConfigs {
     cluster_config: UserTelemetryClusterConfig,
 }
 
+#[derive(Serialize, Clone, Default)]
+pub struct UserTelemetryWebData {
+    responses: HashMap<usize, usize>,
```
Comment (agourlay) [inline src/common/user_telemetry.rs:81]: If the always key represents HTTP status code, maybe we could use a smaller type to represent it.
Maybe `u8` could be enough?

**Review by generall — APPROVED** [review]

**Pull request merged**